### PR TITLE
Zc - Code Size Reduction chapter integration.

### PR DESCRIPTION
**Pull request by wmat**
This PR pull the Zc - Code Size Reduction specification in as a chapter.

### CODE REVIEW

File: src/zc/Zc.adoc [hunk 1]
```diff
@@ -0,0 +1,396 @@
+//:sectnums:
+//:version-label: v1.0.4-2
+//:lifecycle-state: ratified
+
+[#Zc]
+== Zc* {version-label}
+
+=== Change history since v0.70.1 (tagged release)
+
+.Change history
```
Comment (tariqkurd-repo) [inline src/zc/Zc.adoc:10]: do we need the revision history in the unpriv spec? other extensions don't have that

File: src/zc/Zc.adoc [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+Zcd is the existing set of compressed double precision floating point loads and stores: _c.fld_, _c.fldsp_, _c.fsd_, _c.fsdsp_.
+
+The Zcd extension depends on the <<Zca>> and D extensions.
+
+[#Zcb]
+=== Zcb
+
+Zcb has simple code-size saving instructions which are easy to implement on all CPUs.
+
+All proposed encodings are currently reserved for all architectures, and have no conflicts with any existing extensions.
```
Comment (tariqkurd-repo) [inline src/zc/Zc.adoc:169]: the word _proposed_ is no longer relevant

File: src/zc/Zc.adoc [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+
+NOTE: Zcb can be implemented on _any_ CPU as the instructions are 16-bit versions of existing 32-bit instructions from the application class profile.
```
Comment (tariqkurd-repo) [inline src/zc/Zc.adoc:171]: do we want a reference to profiles from here?

File: src/zc/Zcb_footer.adoc [hunk 4]
```diff
@@ -0,0 +1,12 @@
+
```
Comment (tariqkurd-repo) [inline src/zc/Zcb_footer.adoc:1]: do we still need the footer?

File: src/zc/Zcmb_footer.adoc [hunk 5]
```diff
@@ -0,0 +1,12 @@
+
```
Comment (tariqkurd-repo) [inline src/zc/Zcmb_footer.adoc:1]: do we still need the footer?

File: src/zc/Zcmd.adoc [hunk 6]
```diff
@@ -0,0 +1,22 @@
+[#Zcmd]
```
Comment (tariqkurd-repo) [inline src/zc/Zcmd.adoc:1]: this file shouldn't be here - we never finished analysing this one

File: src/zc/Zcmd_footer.adoc [hunk 7]
```diff
@@ -0,0 +1,12 @@
+
```
Comment (tariqkurd-repo) [inline src/zc/Zcmd_footer.adoc:1]: do we still need the footer?

File: src/zc/Zcmp_footer.adoc [hunk 8]
```diff
@@ -0,0 +1,12 @@
+
```
Comment (tariqkurd-repo) [inline src/zc/Zcmp_footer.adoc:1]: do we still need the footer?

File: src/zc/Zcmpe_footer.adoc [hunk 9]
```diff
@@ -0,0 +1,12 @@
+
```
Comment (tariqkurd-repo) [inline src/zc/Zcmpe_footer.adoc:1]: do we still need the footer?

File: src/zc/c_lbu.adoc [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+[NOTE]
+====
+  _rd'_ and _rs1'_ are from the standard 8-register set x8-x15.
+====
+
+Prerequisites::
+None
+
+32-bit equivalent::
+<<insns-lbu>>
```
Comment (tariqkurd-repo) [inline src/zc/c_lbu.adoc:38]: broken xref

File: src/zc/changes_since_v0.50.adoc [hunk 11]
```diff
@@ -0,0 +1,130 @@
+
```
Comment (tariqkurd-repo) [inline src/zc/changes_since_v0.50.adoc:1]: we don't need this file

File: src/zc/cm_decbnez.adoc [hunk 12]
```diff
@@ -0,0 +1,51 @@
+<<<
```
Comment (tariqkurd-repo) [inline src/zc/cm_decbnez.adoc:1]: we don't need this file

File: src/zc/example.bib [hunk 13]
```diff
@@ -0,0 +1,40 @@
+@inproceedings{riscI-isca1981,
```
Comment (tariqkurd-repo) [inline src/zc/example.bib:1]: why is this here?

File: src/zc/readme.md [hunk 14]
```diff
@@ -0,0 +1,15 @@
+This directory has the latest draft specification for the Zc extensions, without the PDF build.
```
Comment (tariqkurd-repo) [inline src/zc/readme.md:1]: don't need this file

File: src/zc/Zc.adoc (hunk 2 above)
Comment (wmat) [reply src/zc/Zc.adoc]: Deleted from consolidated spec.

File: src/zc/Zcb_footer.adoc (hunk 4 above)
Comment (wmat) [reply src/zc/Zcb_footer.adoc]: I've removed all of the footers from the consolidated chapter.

File: src/zc/Zcmb_footer.adoc (hunk 5 above)
Comment (wmat) [reply src/zc/Zcmb_footer.adoc]: I've removed all of the footers from the consolidated chapter.

File: src/zc/Zcmd_footer.adoc (hunk 7 above)
Comment (wmat) [reply src/zc/Zcmd_footer.adoc]: I've removed all of the footers from the consolidated chapter.

File: src/zc/c_lbu.adoc (hunk 10 above)
Comment (wmat) [reply src/zc/c_lbu.adoc]: How do you want to handle these?  I believe they were placeholders.  There is quite a few.

File: src/zc/cm_decbnez.adoc (hunk 12 above)
Comment (wmat) [reply src/zc/cm_decbnez.adoc]: Removed.

File: src/zc/example.bib (hunk 13 above)
Comment (wmat) [reply src/zc/example.bib]: Remnant from original repo.  Removed.

File: src/zc/readme.md (hunk 14 above)
Comment (wmat) [reply src/zc/readme.md]: Removed.

File: src/zc/c_lbu.adoc (hunk 10 above)
Comment (tariqkurd-repo) [reply src/zc/c_lbu.adoc]: is the intention to eventually have separate instruction pages for each instruction?
if so then eventually the xref will point somewhere. If not then just the name of teh 32-bit instruction would do it

**Comment by tariqkurd-repo** [comment]
Hi - how can I get the unpriv PDF? The artefacts don't seem to be kept?

**Comment by tariqkurd-repo** [comment]
I think I'd remove the version history but otherwise it looks good

**Comment by wmat** [comment]
> Hi - how can I get the unpriv PDF? The artefacts don't seem to be kept?

You can always click on Actions, then on the left click RISC-V ISA Build, then on the right Run workflow dropdown, then change the Branch to Zc and click run workflow.  You'll see the workflow begin to run, once it's done, click on it and you'll see the artifacts.

**Comment by wmat** [comment]
> I think I'd remove the version history but otherwise it looks good, except for the missing xrefs which should probably be removed

I removed the version history yesterday.  I'll remove the xrefs today.

File: src/zc/c_lbu.adoc (hunk 10 above)
Comment (wmat) [reply src/zc/c_lbu.adoc]: I have no idea what the intention is for separate instructions, however, leaving xrefs as they are will require whoever authors the 32-bit instructions to make sure they add the same anchor to make these xrefs work.  I think for now I should remove each 32-bit instruction section.  @tariqkurd-repo would you agree with that?

File: src/zc/Zcmp_footer.adoc (hunk 8 above)
Comment (wmat) [reply src/zc/Zcmp_footer.adoc]: I don't believe we do, as the only footer in the original that is used applies to the Zcb instructions and all it says that the instructions is included in Zcb, which is redundant as the instruction is in the Zcb instruction formats section.

File: src/zc.adoc [hunk 15]
```diff
@@ -0,0 +1,2616 @@
...
+[#Zcmt]
+=== Zcmt
+
+Zcmt adds the table jump instructions and also adds the JVT CSR. The JVT CSR requires a 
+state enable if Smstateen is implemented. See <<csrs-jvt>> for details.
+
+This extension reuses some encodings from _c.fsdsp_.  Therefore it is _incompatible_ with <<Zcd>>,
+ which is included when C and D extensions are both present. 
+
+NOTE: Zcmt is primarily targeted at embedded class CPUs due to implementation complexity. Additionally, it is not compatible with architecture class profiles.
```
Comment (sorear) [inline src/zc.adoc:281]:
```suggestion
NOTE: Zcmt is primarily targeted at embedded class CPUs due to implementation complexity. Additionally, it is not compatible with RVA profiles.
```

The A in RVA stands for Application, not Architecture.

File: src/zc.adoc [hunk 16]
```diff
@@ -0,0 +1,2616 @@
...
+| CSH    | c.sh               | funct6 3+| rs1'    | funct1 | uimm 3+| rs2' 2+| op
+| CU     | c.[sz]ext.*, c.not | funct6 3+| rd'/rs1' 5+| funct5            2+| op
+| CMMV   | cm.mvsa01 cm.mva01s| funct6 3+| r1s'     2+| funct2 3+| r2s'   2+| op
+| CMJT   | cm.jt cm.jalt      | funct6 8+| index                          2+| op
+| CMPP   | cm.push*, cm.pop*  | funct6 2+| funct2 4+| urlist   2+| spimm  2+| op
+|=====================================================================
+
+[NOTE]
+====
+c.mul uses the existing CA format
```
Comment (sorear) [inline src/zc.adoc:324]:
```suggestion
c.mul uses the existing CA format.
```

Comment (wmat) [reply src/zc.adoc:324]: Thanks

File: src/zc.adoc (hunk 15 above)
Comment (wmat) [reply src/zc.adoc:281]: Thanks

**Comment by wmat** [comment]
@tariqkurd-repo I've commented out the 32-bit instruction sections with broken links.  Should those instructions get added, these sections can be restored.

Let me know if we're good to merge this chapter now.  Note tha when I merge it, I will archive the original Zc repository.

**Comment by tariqkurd-repo** [comment]
Hi Bill - can you send me a pdf to review? I can't build it until I get my asciidoctor isntallation fixed, which I can't do right now. However - are the encodings now in the overall table of 16-bit encodings - or will that come later?

**Comment by wmat** [comment]
Sure, here's the PDF., Zc is Chapter 28.  Anything content specific is up
to the original author(s) to add to the spec, which can be done post
merge.  Phase 1 of the merge strategy is to merge the ratified specs as is.

File: src/zc.adoc [hunk 17]
```diff
@@ -0,0 +1,2611 @@
...
+[#Zcb]
+=== Zcb
+
+Zcb has simple code-size saving instructions which are easy to implement on all CPUs.
+
+All encodings are currently reserved for all architectures, and have no conflicts with any existing extensions.
+
+NOTE: Zcb can be implemented on _any_ CPU as the instructions are 16-bit versions of existing 32-bit instructions from the application class profile.
+
+The Zcb extension depends on the <<Zca>> extension.
```
Comment (tariqkurd-repo) [inline src/zc.adoc:130]: This renders as the "section 28.3 extension", I think the xref needs a reftext attribute. The same for the other xrefs to extension names.

File: src/zc.adoc [hunk 18]
```diff
@@ -0,0 +1,2611 @@
...
+],config:{bits:16}}
+....
+
+Description:
+
+This instruction takes the one's complement of _rd'/rs1'_ and writes the result to the same register.
+
+[NOTE]
+====
+  _rd'/rs1'_ is from the standard 8-register set x8-x15.
```
Comment (tariqkurd-repo) [inline src/zc.adoc:976]: the _'s haven't caused italics

File: src/zc.adoc [hunk 19]
```diff
@@ -0,0 +1,2611 @@
...
+----
+
+<<<
+
+[#insns-pushpop,reftext="PUSH/POP Register Instructions"]
+=== PUSH/POP register instructions
+
+These instructions are collectively referred to as PUSH/POP: 
+
+* <<#insns-cm_push>>
```
Comment (tariqkurd-repo) [inline src/zc.adoc:1065]: the reference text here should probably be the instruciton name, this list reads strangely

**Comment by tariqkurd-repo** [comment]
I add a few comments, all related to rendering.
The main one is that every xref to e.g. "The <<zca>> extension" renders as "The Section 28.x extension" which isn't as expected.

File: src/zc.adoc (hunk 18 above)
Comment (wmat) [reply src/zc.adoc:976]: That's due to the theme italicizing admonitions by default.  This is consistent with the rest of the spec.

File: src/zc.adoc (hunk 19 above)
Comment (wmat) [reply src/zc.adoc:1065]: That was from the original spec, but I changed it to the instruction names as per your suggestion.

File: src/zc.adoc (hunk 17 above)
Comment (wmat) [reply src/zc.adoc:130]: Fixed.

**Comment by wmat** [comment]
I've updated as per your suggestions.  I'll merge this spec today and any future changes can be made against riscv-isa-manual's main branch.
